### PR TITLE
add RDS data refresh helm job

### PR DESCRIPTION
**Pull request by sobostion**

### CODE REVIEW

**Review by andrewmooreio — COMMENTED** [review]
Other parameters such as RDS instance name and the secret names will need to be templated using the component name/environment. But given we don't know what they're called yet, not sure what else can be done. This ticket should probably have waited until we have the necessary k8s resources to build on top of in hindsight!

**Review by georgepstaylor — APPROVED** [review]

File: helm/refresh-db/templates/job.yaml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                  key: DATABASE_PASSWORD
+            - name: SRC_DB_HOST
+              valueFrom:
+                secretKeyRef:
+                  name: rds-instance-output
+                  key: RDS_INSTANCE_ADDRESS
+            - name: DST_DB_NAME
+              valueFrom:
+                secretKeyRef:
+                  name: rds-instance-output-{{ .Values.destinationEnvironment }}
```
Comment (georgepstaylor) [inline helm/refresh-db/templates/job.yaml:85]:
```suggestion
                  name: rds-instance-output-{{ .Values.destinationEnvironment }}-{{ .Values.component }}
```

File: helm/refresh-db/templates/job.yaml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                  key: DATABASE_NAME
+            - name: DST_DB_USER
+              valueFrom:
+                secretKeyRef:
+                  name: rds-instance-output-{{ .Values.destinationEnvironment }}
```
Comment (georgepstaylor) [inline helm/refresh-db/templates/job.yaml:90]:
```suggestion
                  name: rds-instance-output-{{ .Values.destinationEnvironment }}-{{ .Values.component }}
```

File: helm/refresh-db/templates/job.yaml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                  key: DATABASE_USERNAME
+            - name: DST_DB_PASS
+              valueFrom:
+                secretKeyRef:
+                  name: rds-instance-output-{{ .Values.destinationEnvironment }}
```
Comment (georgepstaylor) [inline helm/refresh-db/templates/job.yaml:95]:
```suggestion
                  name: rds-instance-output-{{ .Values.destinationEnvironment }}-{{ .Values.component }}
```

File: helm/refresh-db/templates/job.yaml [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                  key: DATABASE_PASSWORD
+            - name: DST_DB_HOST
+              valueFrom:
+                secretKeyRef:
+                  name: rds-instance-output-{{ .Values.destinationEnvironment }}
```
Comment (georgepstaylor) [inline helm/refresh-db/templates/job.yaml:100]:
```suggestion
                  name: rds-instance-output-{{ .Values.destinationEnvironment }}-{{ .Values.component }}
```

File: helm/refresh-db/templates/job.yaml [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+                  key: DATABASE_USERNAME
+            - name: SRC_DB_PASS
+              valueFrom:
+                secretKeyRef:
+                  name: rds-instance-output
+                  key: DATABASE_PASSWORD
+            - name: SRC_DB_HOST
+              valueFrom:
+                secretKeyRef:
+                  name: rds-instance-output
```
Comment (georgepstaylor) [inline helm/refresh-db/templates/job.yaml:80]:
```suggestion
                  name: rds-instance-output-{{ .Values.component }}
```

File: helm/refresh-db/templates/job.yaml [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+                  key: DATABASE_NAME
+            - name: SRC_DB_USER
+              valueFrom:
+                secretKeyRef:
+                  name: rds-instance-output
+                  key: DATABASE_USERNAME
+            - name: SRC_DB_PASS
+              valueFrom:
+                secretKeyRef:
+                  name: rds-instance-output
```
Comment (georgepstaylor) [inline helm/refresh-db/templates/job.yaml:75]:
```suggestion
                  name: rds-instance-output-{{ .Values.component }}
```

File: helm/refresh-db/templates/job.yaml [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+              value: "/home/job"
+            - name: SRC_DB_NAME
+              valueFrom:
+                secretKeyRef:
+                  name: rds-instance-output
+                  key: DATABASE_NAME
+            - name: SRC_DB_USER
+              valueFrom:
+                secretKeyRef:
+                  name: rds-instance-output
```
Comment (georgepstaylor) [inline helm/refresh-db/templates/job.yaml:70]:
```suggestion
                  name: rds-instance-output-{{ .Values.component }}
```

File: helm/refresh-db/templates/job.yaml [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+              memory: 2Gi
+          command:
+            - /bin/entrypoint.sh
+          env:
+            - name: HOME
+              value: "/home/job"
+            - name: SRC_DB_NAME
+              valueFrom:
+                secretKeyRef:
+                  name: rds-instance-output
```
Comment (georgepstaylor) [inline helm/refresh-db/templates/job.yaml:65]:
```suggestion
                  name: rds-instance-output-{{ .Values.component }}
```

File: .github/workflows/delius-core-cp-rds-data-refresh.yaml [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+      - name: Uninstall DB Refresh chart
+        run: helm uninstall refresh-db --ignore-not-found
+
+      - name: DB Refresh
+        working-directory: helm/refresh-db
+        run: |
+          helm install refresh-db . \
+            --set sourceEnvironment=${{ github.event.inputs.source_env }} \
+            --set destinationEnvironment=${{ github.event.inputs.destination_env }} \
+            --set slackWebhookUrl=${{ secrets.SLACK_WEBHOOK_URL }}
```
Comment (georgepstaylor) [inline .github/workflows/delius-core-cp-rds-data-refresh.yaml:90]:
```suggestion
          helm install refresh-db-${{ github.events.inputs.component }} . \
            --set sourceEnvironment=${{ github.event.inputs.source_env }} \
            --set destinationEnvironment=${{ github.event.inputs.destination_env }} \
            --set slackWebhookUrl=${{ secrets.SLACK_WEBHOOK_URL }} \ 
            --set component=${{ github.events.inputs.component }}
```

File: .github/workflows/delius-core-cp-rds-data-refresh.yaml [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+          echo "${{ secrets.KUBE_CERT }}" > ca.crt
+          kubectl config set-cluster ${KUBE_CLUSTER} --certificate-authority=./ca.crt --server=https://${KUBE_CLUSTER}
+          kubectl config set-credentials deploy-user --token=${{ secrets.KUBE_TOKEN }}
+          kubectl config set-context ${KUBE_CLUSTER} --cluster=${KUBE_CLUSTER} --user=deploy-user --namespace=hmpps-delius-core-${{ github.event.inputs.destination_env }}
+          kubectl config use-context ${KUBE_CLUSTER}
+        env:
+          KUBE_CLUSTER: ${{ secrets.KUBE_CLUSTER }}
+
+      - name: Uninstall DB Refresh chart
+        run: helm uninstall refresh-db --ignore-not-found
```
Comment (georgepstaylor) [inline .github/workflows/delius-core-cp-rds-data-refresh.yaml:82]:
```suggestion
        run: helm uninstall refresh-db-${{ github.events.inputs.component }} --ignore-not-found
```

**Review by georgepstaylor — CHANGES_REQUESTED** [review]
As per @andrewmooreio's comment, I think this should do the trick?

**Review by georgepstaylor — APPROVED** [review]

**Pull request merged**